internal/kafka/internal/workers/cluster_mgrs/dynamic_scaleup_mgr.go

```diff
@@ -195,96 +277,204 @@ func (m *DynamicScaleUpManager) processDynamicScaleUpReconcileEvent() error {
+		Region:                p.locator.region,
+		SupportedInstanceType: p.locator.instanceTypeName,
 		MultiAZ:               newClusterMultiAZ,
 		Status:                api.ClusterAccepted,
 		ProviderType:          api.ClusterProviderOCM,
 	}
 
-	err := m.ClusterService.RegisterClusterJob(clusterRequest)
+	if p.dryRun {
+		return nil
+	}
+
+	err := p.clusterService.RegisterClusterJob(clusterRequest)
 	if err != nil {
 		return err
 	}
 
 	return nil
 }
+
+func (p *standardDynamicScaleUpProcessor) enoughCapacitySlackInRegion(summary instanceTypeConsumptionSummary) bool {
+	freeStreamingUnitsInRegion := summary.freeStreamingUnits
+	capacitySlackInRegion := p.instanceTypeConfig.MinAvailableCapacitySlackStreamingUnits
+
+	return freeStreamingUnitsInRegion < capacitySlackInRegion
+}
+
+func (p *standardDynamicScaleUpProcessor) regionLimitReached(summary instanceTypeConsumptionSummary) bool {
+	streamingUnitsLimitInRegion := p.instanceTypeConfig.Limit
+	consumedStreamingUnitsInRegion := summary.consumedStreamingUnits
+
+	if streamingUnitsLimitInRegion == nil {
+		return false
+	}
+
+	return consumedStreamingUnitsInRegion >= *streamingUnitsLimitInRegion
+}
+
+func (p *standardDynamicScaleUpProcessor) freeCapacityForBiggestInstanceSize(summary instanceTypeConsumptionSummary) bool {
+	return summary.biggestInstanceSizeCapacityAvailable
+}
+
+func (p *standardDynamicScaleUpProcessor) ongoingScaleUpAction(summary instanceTypeConsumptionSummary) bool {
+	return summary.ongoingScaleUpAction
+}
+
+// instanceTypeConsumptionSummary contains a consumption summary
+// of an instance in a provider's region
+type instanceTypeConsumptionSummary struct {
+	// maxStreamingUnits is the total capacity for the instance type in
+	// streaming units
+	maxStreamingUnits int
+	// freeStreamingUnits is the free capacity for the instance type in
+	// streaming units
+	freeStreamingUnits int
+	// consumedStreamingUnits is the consumed capacity for the instance type in
+	// streaming units
+	consumedStreamingUnits int
+	// ongoingScaleUpAction designates whether a data plane cluster in the
+	// provider's region, which supports the instance type, is being created
+	ongoingScaleUpAction bool
+	// biggestInstanceSizeCapacityAvailable indicates whether there is capacity
+	// to allocate at least one unit of the biggest kafka instance size of
+	// the instance type
+	biggestInstanceSizeCapacityAvailable bool
+}
+
+// instanceTypeConsumptionSummaryCalculator calculates a consumption summary
+// for the provided supportedInstanceTypeLocator based on the
+// provided KafkaStreamingUnitCountPerClusterList
+type instanceTypeConsumptionSummaryCalculator struct {
+	locator                               supportedInstanceTypeLocator
+	kafkaStreamingUnitCountPerClusterList services.KafkaStreamingUnitCountPerClusterList
+	supportedKafkaInstanceTypesConfig     *config.SupportedKafkaInstanceTypesConfig
+}
+
+// Calculate returns a instanceTypeConsumptionSummary containing a consumption
```

Comment:
I love these comments:

Once we've the implementation in place we could copy-paste comments like these onto the https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/docs/architecture/data-plane-osd-cluster-dynamic-scaling.md

Reply:
+:100:  really makes it clear on what each component is for and makes it easier to follow the different conditions in the methods :)